src/commands/hld/reconcile.ts

```diff
@@ -223,3 +251,164 @@ const execAndLog = async (commandToRun: string) => {
+  );
+};
+
+export const createRingComponent = async (
+  execCmd: (commandToRun: string) => Promise<void>,
+  svcPathInHld: string,
+  ring: string
+) => {
+  const createRingInSvcCommand = `cd ${svcPathInHld} && mkdir -p ${ring} config && fab add ${ring} --path ./${ring} --method local --type component && touch ./config/common.yaml`;
+  await execCmd(createRingInSvcCommand);
+};
+
+export const addChartToRing = async (
+  execCmd: (commandToRun: string) => Promise<void>,
+  ringPathInHld: string,
+  serviceConfig: IBedrockServiceConfig
+) => {
+  let addHelmChartCommand = "";
+  const { chart } = serviceConfig.helm;
+  if ("git" in chart) {
+    const chartVersioning =
+      "branch" in chart ? `--branch ${chart.branch}` : `--version ${chart.sha}`;
+    addHelmChartCommand = `fab add chart --source ${chart.git} --path ${chart.path} ${chartVersioning}`;
+  } else if ("repository" in chart) {
+    addHelmChartCommand = `fab add chart --source ${chart.repository} --path ${chart.chart}`;
+  }
+
+  await execCmd(`cd ${ringPathInHld} && ${addHelmChartCommand}`);
+};
+
+export const createStaticComponent = async (
+  execCmd: (commandToRun: string) => {},
+  ringPathInHld: string
+) => {
+  const createConfigAndStaticComponentCommand = `cd ${ringPathInHld} && mkdir -p config static && fab add static --path ./static --method local --type static && touch ./config/common.yaml`;
+  await execCmd(createConfigAndStaticComponentCommand);
+};
+
+export const validateInputs = (
+  repositoryName: any,
+  hldPath: any,
+  bedrockApplicationRepoPath: any
+) => {
+  if (typeof repositoryName !== "string") {
+    throw new Error(
+      `repository-name must be of type 'string', ${typeof repositoryName} given`
+    );
+  }
+
+  if (typeof hldPath !== "string") {
+    throw new Error(
+      `hld-path must be of type 'string', ${typeof hldPath} given`
+    );
+  }
+
+  if (typeof bedrockApplicationRepoPath !== "string") {
+    throw new Error(
+      `bedrock-application-repo-path must be of type 'string', ${typeof bedrockApplicationRepoPath} given`
+    );
+  }
+};
+
+export const testAndGetAbsPath = (
+  test: (flags: TestOptions, path: string) => boolean,
+  log: (logline: string) => void,
+  possiblyRelativePath: string,
+  pathType: string
+): string => {
+  const absPath = path.resolve(possiblyRelativePath);
+  if (!test("-e", absPath) && !test("-d", absPath)) {
+    throw new Error(`Error: could not validate ${pathType} path.`);
+  }
+  log(`Found ${pathType} at ${absPath}`);
+  return absPath;
+};
+
+export const checkForFabrikate = (which: (path: string) => string) => {
+  const fabrikateInstalled = which("fab");
+  if (fabrikateInstalled === "") {
+    throw new Error(
+      `Error: Fabrikate not installed. Please fetch and install the latest version: https://github.com/microsoft/fabrikate/releases`
```

Comment:
don't think we should prefix the errors with "Error: ". When an `Error` gets serialized via `.toString()` it automatically gets an "Error:" prefix. So this would result in `Error: Error: Fabrikate not installed...`

Reply:
makes sense - will change